### PR TITLE
Add Java synthetic property references

### PR DESCRIPTION
**Pull request by PetrakovichVictoria**
[KT-56313](https://youtrack.jetbrains.com/issue/KT-56313) [Docs][Language] Preview of Java synthetic property references

### CODE REVIEW

**Comment by nikitabobko** [comment]
Previous PR: https://github.com/JetBrains/kotlin-web-site/pull/3498

File: docs/topics/jvm/java-interop.md [hunk 1]
```diff
@@ -25,7 +25,8 @@ fun demo(source: List<Int>) {
 ## Getters and setters
 
 Methods that follow the Java conventions for getters and setters (no-argument methods with names starting with `get`
-and single-argument methods with names starting with `set`) are represented as properties in Kotlin.
+and single-argument methods with names starting with `set`) are represented as properties in Kotlin. Such properties are 
+also called _synthetic_.
```
[the remark this answers is not in the document]
Comment (nikitabobko) [reply docs/topics/jvm/java-interop.md:29]: I think it makes sense to mention the whole "synthetic properties" phrase in the definition to make it more clear

**Review by nikitabobko — CHANGES_REQUESTED** [review]

**Review by nikitabobko — APPROVED** [review]

File: docs/topics/jvm/java-interop.md [hunk 2]
```diff
@@ -43,9 +44,88 @@ fun calendarDemo() {
 }
 ```
 
+`calendar.firstDayOfWeek` above is a synthetic property.
```
Comment (sarahhaggarty) [inline docs/topics/jvm/java-interop.md:47]:
```suggestion
`calendar.firstDayOfWeek` above is an example of a synthetic property.
```

File: docs/topics/jvm/java-interop.md [hunk 3]
```diff
@@ -43,9 +44,88 @@ fun calendarDemo() {
...
+
 Note that, if the Java class only has a setter, it isn't visible as a property in Kotlin because Kotlin doesn't support
 set-only properties.
 
+## Java synthetic property references
+
+> This feature is [Experimental](components-stability.md#stability-levels-explained). It may be dropped or changed at any time.
+> Use it only for evaluation purposes. We'd appreciate your feedback on it in [YouTrack](https://kotl.in/issue).
```
Comment (sarahhaggarty) [inline docs/topics/jvm/java-interop.md:55]:
```suggestion
> We recommend that you use it only for evaluation purposes.
```
Danil asked me to not request feedback via YouTrack on experimental features in Docs. Only in what's new. So I guess the same applies here :)

File: docs/topics/jvm/java-interop.md [hunk 4]
```diff
@@ -43,9 +44,88 @@ fun calendarDemo() {
...
+         // Call a reference to Java synthetic property:
+        .sortedBy(Person::age)
+         // Call Java getter via the Kotlin property syntax:
+        .forEach { person -> println(person.name) }
+}
+```
+
+### How to enable Java synthetic property references {initial-collapse-state="collapsed"}
+
+To try this feature out, enable the `-language-version 1.9` compiler option. In a Gradle project, you can do so
```
Comment (sarahhaggarty) [inline docs/topics/jvm/java-interop.md:96]:
```suggestion
To enable this feature, set the `-language-version 1.9` compiler option. In a Gradle project, you can do so 
```

**Review by sarahhaggarty — COMMENTED** [review]
Looking good! Again, just some minor suggestions. 😃

**Pull request merged**